src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -2535,3 +2527,63 @@ ShouldSkipMetadataChecks(void)
 
 	return false;
 }
+
+
+#include "distributed/resource_lock.h"
+
+/*
+ * citus_internal_update_placement_metadata is an internal UDF to
+ * update a row in pg_dist_placement.
+ */
+Datum
+citus_internal_update_placement_metadata(PG_FUNCTION_ARGS)
+{
+	int64 shardId = PG_GETARG_INT64(0);
+	int32 sourceGroupId = PG_GETARG_INT32(1);
+	int32 targetGroupId = PG_GETARG_INT32(2);
+
+	ShardPlacement *placement = NULL;
+	if (!ShouldSkipMetadataChecks())
+	{
+		/* this UDF is not allowed allowed for executing as a separate command */
+		EnsureCoordinatorInitiatedOperation();
+
+		if (!ShardExists(shardId))
+		{
+			ereport(ERROR, (errcode(ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE),
+							errmsg("Shard id does not exists: %ld", shardId)));
+		}
+
+		EnsureShardOwner(shardId);
+
+		placement = ActiveShardPlacementOnGroup(sourceGroupId, shardId);
```

Comment:
I think this can be be moved after the `if` (and then remove the `else` completely)

Reply:
added a clarifying comment